gateway-types/src/ignition/v2.rs

```diff
@@ -0,0 +1,151 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::component::SpIdentifier;
+
+#[derive(Deserialize, JsonSchema)]
+pub struct PathSpIgnitionCommand {
+    /// ID for the SP that the gateway service translates into the appropriate
+    /// port for communicating with the given SP.
+    #[serde(flatten)]
+    pub sp: SpIdentifier,
+    /// Ignition command to perform on the targeted SP.
+    pub command: IgnitionCommand,
+}
+
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Deserialize,
+    Serialize,
+    JsonSchema,
+)]
+pub struct SpIgnitionInfo {
+    pub id: SpIdentifier,
+    pub details: SpIgnition,
+}
+
+/// Ignition command.
```

Comment:
Maybe a little more information for this one? 😄

Reply:
Hah you are finding all the places that were not as well documented